scripts/compare-deposits/compare-deposits.go

```diff
@@ -0,0 +1,242 @@
+	if err != nil {
+		return err
+	}
+
+	now := uint64(time.Now().Unix()) * 1000 // Milliseconds.
+
+	var (
+		deposits []deposit
+		mints    []mint
+	)
+	for i := 0; ; i++ {
+		var from uint64
+		var limit = 100
+		trans, err := cMain.GetNEP17Transfers(fsCont, &from, &now, &limit, &i)
+		if err != nil {
+			return fmt.Errorf("can't get transfers: %w", err)
+		}
+		for _, d := range trans.Received {
+			amount, err := fixedn.FromString(d.Amount, 8)
+			if err != nil {
+				return fmt.Errorf("amount conversion error: %w", err)
+			}
+			deposits = append(deposits, deposit{d.Address, amount, d.TxHash})
+		}
+		if len(trans.Received)+len(trans.Sent) < 100 {
+			break
+		}
+	}
+	slices.Reverse(deposits)
+
+	gasR := gas.NewReader(invoker.New(cMain, nil))
+	bMain, err := gasR.BalanceOf(fsCont)
+	if err != nil {
+		return err
+	}
+	fmt.Println(len(deposits), "deposits, main balance:", fixedn.ToString(bMain, 8))
+
+	nnsHash, err := nns.InferHash(cFS)
+	if err != nil {
+		return err
+	}
+	nnsR := nns.NewReader(invoker.New(cFS, nil), nnsHash)
+	balanceHash, err := nnsR.ResolveFSContract("balance")
+	if err != nil {
+		return err
+	}
+
+	maxH, err := cFS.GetBlockCount()
+	if err != nil {
+		return err
+	}
+	maxH-- // blockCount to height
+
+	var (
+		maxSupply   = getSupply(cFS, balanceHash, maxH)
+		nextSupply  int64
+		knownBlocks = make(map[uint32]int64)
+	)
+
+	fmt.Println("FS supply:", fixedn.ToString(big.NewInt(maxSupply), 12))
+	for nextSupply < maxSupply {
+		var (
+			minBlock  uint32
+			maxBlock  = maxH
+			tmpSupply int64
+		)
+		for h, s := range knownBlocks {
+			if s < nextSupply && minBlock < h {
+				minBlock = h
+			}
+			if s > nextSupply && maxBlock > h {
+				maxBlock = h
+			}
+		}
+		n := sort.Search(int(maxBlock-minBlock), func(i int) bool {
+			var h = minBlock + uint32(i)
+			s := getSupply(cFS, balanceHash, h)
+			knownBlocks[h] = s
+			return s > nextSupply
+		})
+		for _, s := range knownBlocks {
+			if s > nextSupply && (s < tmpSupply || tmpSupply == 0) {
+				tmpSupply = s
+			}
+		}
+		nextSupply = tmpSupply
+		n = int(minBlock) + n
+		fmt.Println("FS block", n, "supply", fixedn.ToString(big.NewInt(nextSupply), 12))
+		b, err := cFS.GetBlockByIndex(uint32(n))
+		if err != nil {
+			return err
+		}
+		for _, t := range b.Transactions {
+			l, err := cFS.GetApplicationLog(t.Hash(), nil)
+			if err != nil {
+				return err
+			}
+			for _, e := range l.Executions[0].Events {
+				if e.ScriptHash.Equals(balanceHash) && e.Name == "TransferX" {
+					itms := e.Item.Value().([]stackitem.Item)
+					_, ok := itms[0].(stackitem.Null)
+					if !ok { // Non-mint.
+						continue
+					}
+					var to util.Uint160
+					toB, _ := itms[1].TryBytes()
+					if len(toB) == 20 {
+						to, _ = util.Uint160DecodeBytesBE(toB)
+					}
+					amount, _ := itms[2].TryInteger()
+					data, _ := itms[3].TryBytes()
+					var h util.Uint256
+					if len(data) == 33 {
+						h, _ = util.Uint256DecodeBytesBE(data[1:33])
```

Comment:
Just out of curiosity, what is the first byte in this case?

Reply:
01 IIRC.